src/lib/util/getOSItemsArray.ts

```diff
@@ -0,0 +1,16 @@
+import { Items } from 'oldschooljs';
+import { Item, PartialItem } from 'oldschooljs/dist/meta/types';
+
+import cleanItemName from './cleanItemName';
+
+export default function getOSItemsArray(itemName: string | number): Item[] | PartialItem[] {
+	const searchedItem = typeof itemName === 'number' ? itemName : cleanItemName(itemName);
+	const osItems = Items.filter(i => {
+		if (typeof searchedItem === 'string') {
+			return i.name.toLowerCase() === searchedItem.toLowerCase();
+		}
+		return i.id === searchedItem;
+	}).array() as Item[];
+	if (!osItems) throw `${itemName} doesnt exist.`;
```

Comment:
Shouldn't this be `!osItems.length`?

Empty arrays are truthy values, so `![]` is `false`. This condition will never be hit as `osItems` (if we're to believe the type signatures) will never be `null` or `undefined`.

Reply:
is that what filter does when it doesnt find anything? returns empty array instead of undefined?

Reply:
Yes, `filter` will always return an empty array if nothing matches the predicate.

I glanced at the [implementation](https://github.com/gc/oldschooljs/blob/master/src/structures/Collection.ts#L254-L262) and it will always return a valid collection type, never `null` or `undefined`.

Reply:
thanks for the heads up, pushed the change now